pkg/abaputils/abaputils_test.go

```diff
@@ -330,4 +330,24 @@ func TestHandleHTTPError(t *testing.T) {
 		assert.EqualError(t, err, fmt.Sprintf("%s: %s - %s", errorValue, abapErrorCode, abapErrorMessage))
 		log.Entry().Info(err.Error())
 	})
+
+	t.Run("Non JSON Error", func(t *testing.T) {
+
+		errorValue := "Received Error"
+		bodyString := `Error message`
+		body := []byte(bodyString)
+
+		resp := http.Response{
+			Status:     "400 Bad Request",
+			StatusCode: 400,
+			Body:       ioutil.NopCloser(bytes.NewReader(body)),
+		}
+		receivedErr := errors.New(errorValue)
+		message := "Custom Error Message"
+
+		err := HandleHTTPError(&resp, receivedErr, message, ConnectionDetailsHTTP{})
```

Comment:
I don't quite understand the test. Why does `HandleHTTPError()` have an error as one of its arguments? Is this an error that the function should return in case of *any* failure? If that is the intention, then it obscures the actual errors.